### ISSUE TITLE
Different test-loss with test_setup.sh

### ISSUE THREAD

**Issue by kosmasgal** [YalaLab/pillar-finetune#3]
Hi,

after an uneventful installation I have encountered a difference in the sample loss. 
Here are the details:

## Expected vs observed

- **README:** test loss **0.5797**.
- **Observed:** test loss **0.6579**. 

## Steps to reproduce

1. Clone repo, install deps (`uv sync`, `uv add flash-attn==2.7.0.post2 --no-build-isolation`), install rad-vision-engine from https://github.com/YalaLab/rave/
2. Run in bash:
   OMP_NUM_THREADS=2 NUM_GPUS=1 CUDA_VISIBLE_DEVICES=0 MASTER_PORT=2300 bash scripts/test_setup.sh

### Full test output logs/csv/seed0/checkpoints/3/test.csv:
```
survival,time_at_event,y
[-2.21875 -2.21875 -2.21875 -2.21875 -2.21875 -2.21875],3,1
```

Is such a discrepancy expected? Was the output in the README.md perhaps created using an earlier model?
   
Thank you for you hard work and for making it all public.

**Comment by longchaoliu** [comment]
Hello, thank you for pointing this out! We are investigating this discrepancy, which we can reproduce on our end. We'll let you know what we find.

**Comment by longchaoliu** [comment]
Hello, please pull the latest version of the UV environment, which uses flash_attn 2.8.3: https://github.com/YalaLab/pillar-finetune/commit/47506d5267cb448867fe415d416910ce6aad9422. The discrepancy should be resolved now. Thanks for bearing with us, and let us know if it works!

**Comment by kosmasgal** [comment]
Hi Joy! Thank you very much for looking into it. Unfortunately I am still getting test_loss = 0.6579 even after upgrading to flash-attn==2.8.3 (confirmed installed). Same result with 2.7.0.post2 and 2.8.3 -- so the version bump doesn't seem to be the fix.

My setup matches origin/main exactly for the test_setup.sh code path (verified via git diff). Checkpoint md5: 9de5c2c1146d17cae66117aed5a873bc.

Could the discrepancy be GPU-architecture dependent? What GPU / CUDA / torch versions were used to generate the README result (0.5797)? My env: RTX A6000, CUDA 12.8, torch 2.8.0+cu128.
